src/tax/basis.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+
+import moment from "moment";
+
+export type BasisFrame = {
+  date: moment,
+  amount: number,
+  cost: number,
+};
+
+/** A cost basis calculator with LIFO (Last In, First Out) semantics.
+ * Any time that a currency is being disposed of (e.g. sold or gifted),
+ * it reports the effective cost basis for that amount of currency, by
+ * returning the appropriate BasisFrames. Note that it may return several
+ * BasisFrames, e.g. if you acquired 50 at one cost, 50 at another cost, and then
+ * dispose of 75, you will get 2 frames (50 from the last frame, 25 from the first one).
+ */
+export class LIFOCostBasisCalculator {
+  frames: BasisFrame[];
+  ticker: string;
+
+  constructor(ticker: string) {
+    this.frames = [];
+    this.ticker = ticker;
+  }
+
+  /** Record that a cryptocurrency was acquired at a certain cost */
+  acquire(date: moment, amount: number, cost: number) {
```

Comment:
Suggest accepting a `frame: BasisFrame` here instead of these three parameters. Already while reading the tests I was confused at some call to `acquire` because I forgot the order of the two `number` parameters. (You should shallow-copy the frame before pushing it, though: `this.frames.push({...frame})`.)